mongoengine/fields.py

```diff
@@ -619,6 +620,14 @@ class DynamicField(BaseField):
 
     Used by :class:`~mongoengine.DynamicDocument` to handle dynamic data"""
 
+    def __init__(self, container_class=dict, *args, **kwargs):
+        self._container_cls = container_class
+        if not issubclass(self._container_cls, Mapping):
+            self.error('The class that is specified in `container_class` parameter '
```

Comment:
This shouldn't throw a `ValidationError`, which is reserved to invalid *data*, not an invalid field definition.